common/djangoapps/enrollment/data.py

```diff
@@ -144,7 +144,7 @@ def update_course_enrollment(username, course_id, mode=None, is_active=None):
         A serializable dictionary representing the modified course enrollment.
 
     """
-    course_key = CourseKey.from_string(course_id)
+    course_key = course_key_from_string_or_404(course_id)
```

Comment:
is this an appropriate place to call this helper? is this function only called by a view?